waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -240,15 +242,17 @@
   "Make an HTTP request to the Kubernetes API server using the configured authentication.
    If data is provided via :body, the application/json content type is added automatically.
    The response payload (if any) is automatically parsed to JSON."
-  [client url & {:keys [body content-type request-method] :as options}]
+  [client url {:keys [scheduler-name]} & {:keys [body content-type request-method] :as options}]
   (scheduler/log "making request to K8s API server:" url request-method body)
   (ss/try+
     (let [auth-str @k8s-api-auth-str
-          result (pc/mapply http-utils/http-request client url
-                            :accept "application/json"
-                            (cond-> options
-                              auth-str (assoc-in [:headers "Authorization"] auth-str)
-                              (and (not content-type) body) (assoc :content-type "application/json")))]
+          result (timers/start-stop-time!
+                   (metrics/waiter-timer "scheduler" scheduler-name request-method)
```

Comment:
`request-method` may be `nil` here. Maybe we should set the default here explicitly. Currently the default is set to `:get` in the call to `http-utils/http-request` below.